common/build/eslint-plugin-fluid/src/rules/no-unchecked-record-access.js

```diff
@@ -515,3 +578,111 @@ function isStrictlyTypedVariable(typeAnnotation) {
 	// Consider any non-union type as strictly typed, except for 'any' and 'unknown'
 	return typeAnnotation.type !== "TSAnyKeyword" && typeAnnotation.type !== "TSUnknownKeyword";
 }
+
+// Helper function to find the containing block (e.g., BlockStatement)
+function findContainingBlock(node) {
+	let current = node;
+	while (current) {
+		if (current.type === "BlockStatement" || current.type === "Program") {
+			return current;
+		}
+		current = current.parent;
+	}
+	return null;
+}
+
+// Helper to resolve variable values in scope
+function getKeyValue(node, context) {
+	if (node.type === "Literal") return node.value;
+	if (node.type === "Identifier") {
+		let scope = context.getScope();
+		while (scope) {
+			const variable = scope.variables.find((v) => v.name === node.name);
+			if (variable) {
+				// Check all definitions for initial values
+				for (const def of variable.defs) {
+					const init = def?.node?.init;
+					if (!init) continue;
+
+					// Handle nested literals (e.g., const key = "test")
+					if (init.type === "Literal") {
+						return init.value;
+					}
+					// Handle simple identifier references (e.g., const key = KEY_CONSTANT)
+					if (init.type === "Identifier") {
+						return getKeyValue(init, context); // Resolve recursively
+					}
+				}
+				break;
+			}
+			scope = scope.upper;
+		}
+		return node.name;
```

Comment:
Not sure I understand this... what does it mean that we end up returning a `node.name` as the "value" of something? The returns further up I think I get better, like returning the value used to initialize a variable (`init.value`).

Reply:
Yea this return value does seem unclear. When we can't resolve a variable to a literal value (either directly or through reference), we return the variable name itself. This allows us to track the unresolved identifier for validation purposes further up the chain. I've also added these comments